crates/cli_utils/src/helpers.rs

```diff
@@ -188,6 +131,7 @@ where
     I: IntoIterator<Item = S>,
     S: AsRef<OsStr>,
 {
+    let roc_path = build_roc_bin_cached();
```

Comment:
Fundamentally, this is the only real change. Instead of passing a path into this function, we now grab the path/build roc here. Now, all calls will be forced to build roc if it doesn't exist. I also, shifted some code around, but no other functionality should be changed.